libbeat/common/mapstr.go

```diff
@@ -57,6 +58,110 @@ func (m MapStr) Update(d MapStr) {
 	}
 }
 
+func (m MapStr) Delete(key string) error {
+	keyParts := strings.Split(key, ".")
+	keysLen := len(keyParts)
+
+	mapp := m
+	for i := 0; i < keysLen-1; i++ {
+		keyPart := keyParts[i]
+
+		if _, ok := mapp[keyPart]; ok {
+			mapp, ok = mapp[keyPart].(MapStr)
+			if !ok {
+				return fmt.Errorf("unexpected type of %s key", keyPart)
+			}
+		} else {
+			return fmt.Errorf("unknown key %s", keyPart)
+		}
+	}
+	delete(mapp, keyParts[keysLen-1])
+	return nil
+}
+
+func (m MapStr) CopyTo(to MapStr, key string) error {
```

Comment:
Maybe a better name for this function would be `CopySubtree`? Alternatively a godoc comment about what it does would be good.